File: easy-solution/galaxy4276/141.java
class ListNode {
  int val;
  ListNode next;
  ListNode(int x) {
    val = x;
    next = null;
  }
}

class Solution {
  public boolean hasCycle(ListNode head) {
    if (head == null) return false;
    ListNode slow = head;
    ListNode fast = head;
    while (fast != null && fast.next != null && fast.next.next != null) {
      fast = fast.next.next;
      slow = slow.next;
      if (fast == slow) return true;
    }
    return false;
  }
}

public class Main {
  public static void main(String[] args) {
    Solution s = new Solution();
    ListNode head = new ListNode(3);
    ListNode two = new ListNode(2);
    head.next = two;
    ListNode third = new ListNode(0);
    two.next = third;
    ListNode fourth = new ListNode(-4);
    third.next = fourth;
    fourth.next = two;
    boolean result = s.hasCycle(head);
    System.out.println("result: " + result);
  }
}

File: easy-solution/minesp3164/141.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        if head:
            slow = head
            fast = head.next
            while (slow != fast):
                if fast or fast.next:
                    slow = slow.next
                    fast = fast.next.next
                else:
                    return False
            return True
        return False
